Kudu.Services/Diagnostics/ProfileManager.cs

```diff
@@ -0,0 +1,314 @@
+﻿using Kudu.Contracts.Tracing;
+using Kudu.Core.Infrastructure;
+using Kudu.Core.Tracing;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace Kudu.Services.Performance
+{
+    internal static class ProfileManager
```

Comment:
My point is that we have an established pattern for services, and this ProfileManager is another service peer to ITriggeredJobManager, IContinuousJobManager, etc. So I'd make it follow the same model - it's an instance type with instance methods, and a singleton instance is added to the DI container on startup. With an IProfileManager interface (like all our other services), the controller becomes much more easily testable.